src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/SlidingWindow.java

```diff
@@ -23,59 +23,74 @@
  * This is a generic sliding window for RCA sampling. The default behavior is to store the
  * {timestap, value} pair and maintain the sum of all data entries within this sliding window.
  */
-public class SlidingWindow {
+public class SlidingWindow<E extends SlidingWindowData> {
 
-  protected final Deque<SlidingWindowData> windowDeque;
-  protected final int SLIDING_WINDOW_SIZE_IN_SECONDS;
+  protected final Deque<E> windowDeque;
+  protected final long SLIDING_WINDOW_SIZE;
   protected double sum;
 
-  public SlidingWindow(int SLIDING_WINDOW_SIZE_IN_SECONDS) {
+  public SlidingWindow(int SLIDING_WINDOW_SIZE_IN_TIMESTAMP, TimeUnit timeUnit) {
     this.windowDeque = new LinkedList<>();
-    this.SLIDING_WINDOW_SIZE_IN_SECONDS = SLIDING_WINDOW_SIZE_IN_SECONDS;
+    this.SLIDING_WINDOW_SIZE = timeUnit.toSeconds(SLIDING_WINDOW_SIZE_IN_TIMESTAMP);
```

Comment:
If we are converting the name unit to seconds, shouldn't the member variable name remain `**_SECONDS` ?